scripts/run_e2e_tests.py

```diff
@@ -0,0 +1,472 @@
+    '--skip-install',
+    help='If true, skips installing dependencies. The default value is false.',
+    action='store_true')
+_PARSER.add_argument(
+    '--sharding', default=True, type=bool,
+    help='Disables/Enables parallelization of protractor tests.'
+         'Sharding must be disabled (either by passing in false to --sharding'
+         ' or 1 to --sharding-instances) if running any tests in isolation'
+         ' (fit or fdescribe).',
+    )
+_PARSER.add_argument(
+    '--sharding-instances', type=int, default=3,
+    help='Sets the number of parallel browsers to open while sharding.'
+         'Sharding must be disabled (either by passing in false to --sharding'
+         ' or 1 to --sharding-instances) if running any tests in isolation'
+         ' (fit or fdescribe).')
+_PARSER.add_argument(
+    '--prod_env',
+    help='Run the tests in prod mode. Static resources are served from'
+         ' build directory and use cache slugs.',
+    action='store_true')
+_PARSER.add_argument(
+    '--community_dashboard_enabled', action='store_true',
+    help='Run the test after enabling the community dashboard page.')
+
+_PARSER.add_argument(
+    '--suite', default='full',
+    help='Performs test for different suites, here suites are the'
+         'name of the test files present in core/tests/protractor_desktop/ and'
+         'core/test/protractor/ dirs. e.g. for the file'
+         'core/tests/protractor/accessibility.js use --suite=accessibility.'
+         'For performing a full test, no argument is required.')
+
+SUBPROCESSES = []
+
+
+def check_screenshot():
+    """Check if screenshot directory exists, if so, delete it."""
+    screeenshots_dir = os.path.join(os.pardir, 'protractor-screenshots')
+    if not os.path.isdir(screeenshots_dir):
+        return
+    python_utils.PRINT(
+        'Note: If ADD_SCREENSHOT_REPORTER is set to true in'
+        'core/tests/protractor.conf.js, you can view screenshots'
+        'of the failed tests in ../protractor-screenshots/')
+    os.rmdir(screeenshots_dir)
+
+
+def cleanup():
+    """Kill the running subprocesses and server fired in this program."""
+    dev_appserver_path = '%s/dev_appserver.py' % common.GOOGLE_APP_ENGINE_HOME
+    webdriver_download_path = '%s/downloads' % WEBDRIVER_HOME_PATH
+    if common.is_windows_os():
+        # In windows system, the java command line will use absolute path.
+        webdriver_download_path = os.path.abspath(webdriver_download_path)
+    processes_to_kill = [
+        '.*%s.*' % re.escape(dev_appserver_path),
+        '.*%s.*' % re.escape(webdriver_download_path)
+    ]
+    for p in SUBPROCESSES:
+        p.kill()
+
+    for p in processes_to_kill:
+        common.kill_processes_based_on_regex(p)
+
+
+def check_running_instance(*ports):
+    """Check if exists running oppia instance.
+
+    Args:
+        ports: list(int). A list of ports that oppia instances may be running.
+
+    Return:
+        bool: Whether the ports are all open.
```

Comment:
Whether any port has a running Oppia instance. (perhaps?)

That said, I'm a bit confused, because the docstring talks about an "oppia instance" but it looks like this is only checking for whether ports are used in general, in a way that's agnostic to what process is being run on that port. Could the docstring be clarified?

Also the original return value description seems to have a confusion between "all ports are open" (based on the description) and "at least one port is open" (based on the impl). Not sure what is intended here.

Reply:
Updated based on the below comment